activestorage/s3.py

```diff
@@ -42,16 +42,24 @@ def reduce_chunk(server, username, password, source, bucket, object, offset,
     if filters is not None:
         raise NotImplementedError("Filters are not yet supported!")
 
-    request_data = build_request_data(source, bucket, object, offset, size, compression, filters, missing, dtype, shape, order, chunk_selection)
+    request_data = build_request_data(source, bucket, object, offset, size,
+                                      compression, filters, missing, dtype,
+                                      shape, order, chunk_selection)
     api_operation = "sum" if operation == "mean" else operation or "select"
     url = f'{server}/v1/{api_operation}/'
-    response = request(url, username, password, request_data)
+    try:
+        response = request(url, username, password, request_data)
+    except requests.exceptions.ConnectionError as exc:
+        raise OSError(f"Resource unreachable: {str(exc)}")
```

Comment:
OSError doesn't seem right to me. If you want to translate the error, then I suggest creating your own exception hierarchy, probably with a common ActiveStorageError base class and some subclasses. We could move the existing `S3ActiveStorageError` into a new file.

Reply:
Perhaps `request` should translate the error, to keep this function high-level?

Reply:
that is indeed a good point - was just thinking about this myself, but I acted lazy - will create a nice error class and stop being lazy :grin: